pandas/core/dtypes/missing.py

```diff
@@ -406,4 +411,7 @@ def remove_na_arraylike(arr):
     """
     Return array-like containing only true/non-NaN values, possibly empty.
     """
-    return arr[notna(lib.values_from_object(arr))]
+    if is_extension_array_dtype(arr):
```

Comment:
ultimately values_from_object just calls .get_values() on the object if it exists. This smells like something is missing from the EA interface.

Reply:
I don't think this if / elif can be avoided. `values_from_object` gets and calls `arr.get_values` if defined, which returns an ndarray. But we need to call `notna` on the EA itself, since it has special missing value handling.